src/DataWrangling/restoring.jl

```diff
@@ -3,66 +3,66 @@ using Oceananigans.Grids: AbstractGrid, node, on_architecture
 using Oceananigans.Fields: interpolate!, interpolate, location, instantiated_location
 using Oceananigans.OutputReaders: Cyclical, TotallyInMemory, AbstractInMemoryBackend, FlavorOfFTS, time_indices
 using Oceananigans.Utils: Time
+using Oceananigans.Architectures: AbstractArchitecture
 
-using Base
-using NCDatasets
 using JLD2
+using NCDatasets
 
 using Dates: Second
 using ClimaOcean: stateindex
 using ClimaOcean.DataWrangling: NearestNeighborInpainting, native_times, default_inpainting, empty_field
 
 import Oceananigans.Fields: set!
 import Oceananigans.Forcings: regularize_forcing
-import Oceananigans.OutputReaders: new_backend, update_field_time_series!
+import Oceananigans.OutputReaders: new_backend, update_field_time_series!, FieldTimeSeries
 
 @inline instantiate(T::DataType) = T()
 @inline instantiate(T) = T
 
-struct ECCONetCDFBackend{N, C, I, M} <: AbstractInMemoryBackend{Int}
+struct NetCDFBackend{N, C, I, M} <: AbstractInMemoryBackend{Int}
```

Comment:
```suggestion
struct DatasetNetCDFBackend{N, C, I, M} <: AbstractInMemoryBackend{Int}
```

I suggest this because we will eventually need to have a "NetCDFBackend" for other types of netcdf files --- eg ones created by Oceananigans. In this case, we probably don't want to re-use the concept of "dataset" (which I think we would like to apply specifically to external / established data products that can be downloaded). 

I'm flexible on the name, just think that we need to be more specific than just "NetCDF"

Reply:
Gotcha! I think makes sense what you suggest!